Add newlines, large benchmark, and throughput measures to benchmarks

CeleritasCelery opened the pull request:
Added a throughput measurement to the criterion report. This will make it easier to see GB/s for a particular algorithm. I like looking at throughput because you can compare benchmarks regardless of size. 

Also added a large benchmark that will test crossing chunk boundaries. These should be near the maximum throughput.

 Lastly added the following line separators to the benchmarks text:

- Line Feed
- Vertical Tab
- Form Feed
- Carriage Return
- Line Separator
- Next Line
- Paragraph Separator

The benchmark names no longer exactly match their size. I am not sure if this a problem or not. Let me know what you think of the changes.

cessen commented:
> The benchmark names no longer exactly match their size. I am not sure if this a problem or not.

Definitely not a problem!  As long as they're within a few percent, it's fine.  Especially since you're listing the throughput now anyway.

Having said that, it occurs to me that this probably isn't really exercising the line functions very well, since the line breaks are so infrequent.  I wonder if it might make more sense to leave the existing benchmark texts alone, and instead create new texts (probably just 100-ish bytes, which we can extend at run time with `repeat()`) with really frequent line breaks, and corresponding new benchmarks just for the line conversion functions.

CeleritasCelery committed:
Add newlines, large benchmark, and throughput measures to benchmarks

Added a throughput measurement to the cirterion report. This will make it easier
to measure GB/s for a particular algorithm. Also added a large benchmark that
will test crossing chunk boundaries. Lastly added the following line separators
to the benchmarks:

- Line Feed
- Vertical Tab
- Form Feed
- Carriage Return
- Line Separator
- Next Line
- Paragraph Separator

CeleritasCelery commented:
Let me know what you think of this change. I reverted the old text files and added a new one specifically for lines. It will now test each type of line ending independently.

cessen commented:
Yes, this looks great!  Thanks!
